src/components/CountryPage/BorderCountries/BorderCountries.jsx

```diff
@@ -1,50 +1,46 @@
 import { useContext, useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import { toast } from "react-toastify";
-import { DarkModeContext } from "../../../contexts/DarkModeContext";
+import { DarkModeContext } from "../../../contexts/DarkModeContext/DarkModeContext";
 import {
   BorderCountriesContainer,
   BorderCountryLink,
 } from "./BorderCountries.styles";
-import ToastComponent from "../../ToastComponent/ToastComponent";
+import useFetchBorderCountriesNames from "./useFetchBorderCountriesNames";
+import ToastComponent from "../../ToastComponent";
+import Spinner from "../../Spinner";
 
 export default function BorderCountries({ borderCountries }) {
   const { isDarkMode } = useContext(DarkModeContext);
-  const [fullNames, setFullNames] = useState([]);
-  const fetchBorderCountriesNames = async () => {
-    try {
-      const response = await fetch(
-        `https://restcountries.com/v2/alpha?codes=${borderCountries}`
-      );
-      if (response.ok) {
-        const data = await response.json();
-        setFullNames(data);
-      } else {
-        throw response.status;
-      }
-    } catch (err) {
+  const { names, loading, error } =
+    useFetchBorderCountriesNames(borderCountries);
+
+  useEffect(() => {
+    if (error) {
       toast(
-        `Unexpected problem occurred(${err}). Cannot fetch border countries. Please try again later.`
+        `Unexpected problem occurred(${error}). Cannot fetch country's details. Please try again later.`
```

Comment:
tu tak samo - toast może być jako fragment funkcji pobierającej (dorzuć to w bloku catch)